src/mo_def/arrange.ml

```diff
@@ -54,10 +54,17 @@ module Make (Cfg : Config) = struct
   | Mo_types.Type.Triv -> Atom "Triv"
   | Mo_types.Type.Await -> Atom "Await"
 
-  let annot_typ t it = if Cfg.include_types then ":" $$ [it; typ t] else it
+  let annot_typ t it =
+    if Cfg.include_types
+    then
+      if Cfg.include_type_rep
+      then ":" $$ [it; typ t; Arrange_type.typ t]
```

Comment:
Can't these type annotations get huge? There's no sharing at all so every subexpression will receive its own type annotation that will duplicate lots of information that could be shared with other type annotations.

Reply:
The current implementation introduces `typ t` to every node, which is the type AST in string form, so we thought it would be fine to also introduce the AST in its structural form, provided that both `include_types` and `include_type_rep` are enabled. But after investigating a bit, it's indeed a problem that can be observed in benchmarks.

Perhaps one simple way to mitigate this might be to arrange these types only where the language server currently requires it, I'll better investigate this point and get back to you.